libs/contrib/System/Path.idr

```diff
@@ -147,203 +151,244 @@ pathTokenMap = toTokenMap $
+|||   Windows only).
+||| - Repeated separators are ignored, therefore, "a/b" and "a//b" both have "a"
 |||   and "b" as bodies.
-||| - Occurrences of "." are normalized away, except if they are at the
-|||   beginning of the path. For example, "a/./b", "a/b/", "a/b/". and 
-|||   "a/b" all have "a" and "b" as bodies, but "./a/b" starts with an 
-|||   additional `CurDir` body.
+||| - "." in the body are removed, unless they are at the beginning of the path.
+|||   For example, "a/./b", "a/b/", "a/b/." and "a/b" will have "a" and "b" as
+|||   bodies, and "./a/b" will starts with `CurDir`.
 |||
 ||| ```idris example
 ||| parse "C:\\Windows/System32"
-||| ```
-||| ```idris example
 ||| parse "/usr/local/etc/*"
 ||| ```
 export
 parse : String -> Path
-parse str = case parse parsePath (lexPath str) of
-                 Right (p, _) => p
-                 _ => emptyPath
+parse str =
+  case parse parsePath (lexPath str) of
+    Right (path, _) => path
+    _ => emptyPath
 
 --------------------------------------------------------------------------------
 -- Utils
 --------------------------------------------------------------------------------
 
 isAbsolute' : Path -> Bool
-isAbsolute' p = if isWindows
-                  then case p.volume of
-                            Just (UNC _ _) => True
-                            Just (Disk _) => p.hasRoot
-                            Nothing => False
-                  else p.hasRoot
+isAbsolute' path =
+  if isWindows then
+    case path.volume of
+      Just (UNC _ _) => True
+      Just (Disk _) => path.hasRoot
+      Nothing => False
+  else
+    path.hasRoot
 
 append' : (left : Path) -> (right : Path) -> Path
-append' l r = if isAbsolute' r || isJust r.volume
-                 then r
-                 else if hasRoot r
-                         then record { volume = l.volume } r
-                         else record { body = l.body ++ r.body,
-                                       hasTrailSep = r.hasTrailSep } l
+append' left right =
+  if isAbsolute' right || isJust right.volume then
+    right
+  else if hasRoot right then
+    record { volume = left.volume } right
+  else 
+    record { body = left.body ++ right.body, hasTrailSep = right.hasTrailSep } left
+
+splitPath' : Path -> List Path
+splitPath' path =
+  case splitRoot path of
+    (Just root, other) => root :: iterateBody (path.body) (path.hasTrailSep)
+    (Nothing, other) => iterateBody (path.body) (path.hasTrailSep)
+  where
+    splitRoot : Path -> (Maybe Path, Path)
+    splitRoot path@(MkPath Nothing False _ _) = (Nothing, path)
+    splitRoot (MkPath vol root xs trailSep) =
+      (Just $ MkPath vol root [] False, MkPath Nothing False xs trailSep)
+
+    iterateBody : List Body -> (trailSep : Bool) -> List Path
+    iterateBody [] _ = []
+    iterateBody [x] trailSep = [MkPath Nothing False [x] trailSep]
+    iterateBody (x::y::xs) trailSep =
+      (MkPath Nothing False [x] False) :: iterateBody (y::xs) trailSep
```

Comment:
isn't this `map (\x => MkPath Nothing False [x] False trailSep)` ?

Reply:
In the case of `iterateBody (x::y::xs) trailSep`, the trailSep should be false. Do we have another way to express that?

Reply:
Ah, I noticed I commended on the wrong file (since Utils.Path is duplicated). my confusion stemmed from this comment https://github.com/idris-lang/Idris2/pull/823/files/9f72a514f1ac523136423caceba725f975a37df5#diff-5621839bbc877aa5fbbbc07903cf1cdf7a3036ecf85836d96de83f873ca1d5a6R296